CONTRIBUTING.md

```diff
@@ -0,0 +1,185 @@
+| docs     | 📜 Documentation | Documentation changes only                                                          | Maybe        |
+| style    | 💅 Style         | Changes that do not affect the meaning of the code (e.g. linting)                   | Maybe        |
+| chore    | ⚙️ Chores        | Changes that do not modify source code (e.g. CI configuration files, build scripts) | No           |
+| revert   | ♻️ Reverts       | Reverts a previous commit                                                           | Maybe        |
+| security | 🔒 Security      | Security fix/feature                                                                | Maybe        |
+
+## Setting up your environment
+
+We use [uv](https://astral.sh/blog/uv) for package and env management. Follow the [uv guide to bootstrap an environment](https://docs.astral.sh/uv/getting-started/installation/), run:
+
+```sh
+curl -LsSf https://astral.sh/uv/install.sh | sh
+```
+
+We use:
+
+1. [`uv`](<(https://astral.sh/blog/uv)>) for package and env management -- follow the [uv guide to bootstrap an environment](https://docs.astral.sh/uv/getting-started/installation/)
+   ```sh
+   curl -LsSf https://astral.sh/uv/install.sh | sh
+   ```
+2. [`nox`](https://nox.thea.codes/en/stable/tutorial.html#installation) for running tests
+   ```sh
+   uv tool install nox
+   ```
+
+### Working with the `weave` package
+
+We recommend installing the package in editable mode:
+
+```sh
+uv pip install -e .
+```
+
+### Linting
+
+We use pre-commit. You can install with:
+
+```sh
+uv tool install pre-commit
+```
+
+Then run:
+
+```sh
+pre-commit run --hook-stage=pre-push --all-files
+```
+
+You can also use the `lint` nox target to run linting.
+
+```sh
+nox -e lint
+```
+
+### Building the `weave` package
+
+Use `uv`:
+
+```sh
+uv build
+```
+
+### Testing
+
+We use pytest and nox to run tests. To see a list of test environments:
+
+```sh
+nox -l
+```
+
+Then to run a specific environment:
+
+```sh
+nox -e "$TARGET_ENV"  # e.g. nox -e "tests-3.12(shard='trace')"
+```
+
+Tests are split up into shards, which include:
+
+1. `trace` -- all of the trace SDK tests
+2. `trace_server` -- tests for the clickhouse trace server backend
+3. various integrations, like `openai`, `instructor`, etc -- these envs are isolated to simplify testing
+
+### Adding settings
+
+Settings are currently handled on the `UserSettings` object in `weave.trace.settings` and via env vars.
+
+To add a new setting:
+
+1. Add a new field to the `UserSettings` class and provide a short description; and
+2. Add a `should_{new_setting_name}` func to the `settings` module so users can check if that setting is enabled
+
+By default, the env var will be `WEAVE_{setting_name_in_all_caps}`, e.g. `WEAVE_PRINT_CALL_LINK`.
+
+### Adding URLs
+
+To keep URLs consistent, make sure to add and access URLs from `weave.trace.urls`. We currently use functions to return URLs.
+
+### Deprecating features
```

Comment:
Does this work? I have never tried it.

Reply:
yeah we use it in a few places, like here:
https://github.com/wandb/weave/blame/andrew/docs4/weave/trace/weave_client.py#L590